### PR TITLE
Cantherm Free Rotors and Bug fixes

### PR DESCRIPTION
**Pull request by mjohnson541**
This pull request replaces #1010, which became disconnected.  Unit tests have been added.

### CODE REVIEW

**Comment by mjohnson541** [comment]
The unit tests, test all of the key FreeRotors functions and test that the proper error is raised if the frequency and geometry files are different in a Gaussian hindered rotor calculation.  I've also added an example (thermo of the benzyl radical) for which this issue originally occurred.  

This should be ready for review now.

**Comment by nyee** [comment]
This is also a continuation of #942 for reference.

File: rmgpy/cantherm/commonTest.py [hunk 1]
```diff
@@ -265,7 +266,20 @@ def testTransitionStateStatmech(self):
...
-
-
+        
+class testStatmech(unittest.TestCase):
+    """
+    Contains unit tests of statmech.py
+    """
+    def setUp(self):
+        cantherm = CanTherm()
+        jobList = cantherm.loadInputFile(os.path.join(os.path.dirname(os.path.abspath(__file__)),r'files/Benzyl/input.py'))
```
Comment (KEHANG) [inline rmgpy/cantherm/commonTest.py]: change `files/Benzyl/input.py` to `test_data/Benzyl/input.py`, just a name convention for test data directory

**Comment by KEHANG** [comment]
Although there's lots of explanations in #981, I want to emphasize what a user should do when dealing with hinder rotor (after this PR merged). Two options to avoid inconsistency in coordinate systems.

- option 1: separate `opt` and `freq` jobs, which generate `opt.log` and `freq.log`. Use `freq.log` for geometry and frequency jobs in cantherm, and only use `opt.log` for energy job.

- option 2: combine `opt` and `freq` in Gaussian input, which generates one log file `comb.log`. Use this log file for all the jobs in cantherm (geometry, frequency, and energy).

@nyee @zjburas @mjohnson541 correct me if I'm summarizing wrongly or if you want to add suggestions for the users.

**Comment by zjburas** [comment]
@KEHANG , there is only one guaranteed way to prepare a cantherm input file if hindered rotors are included: option 1 above. 

Option 2 will not work, because if you combine "opt freq" in one job and try to include the "iop(7/33=1)" keyword, which should output the Hessian, the keyword will only be applied to the first part of the job (opt) but not the second (freq) where it is needed. Therefore, the output of the "opt freq iop(7/33=1)" job will not include the Hessian that is needed for the hindered rotor treatment.

I am working with @alongd on updating the Cantherm documentation to explain all of this, and many other details of using Cantherm that are currently not in the documentation or are out of date. This will likely be in a separate pull request. We should definitely make sure it is clear to future users what the Cantherm input should look like.

File: rmgpy/cantherm/commonTest.py (hunk 1 above)
Comment (mjohnson541) [inline rmgpy/cantherm/commonTest.py]: So when I try to do this I think nosetests starts trying to run the methoxy.py file in the test_data folder or at least it's giving me this error when I run unit tests:
====================================================================
ERROR: Failure: NameError (name 'transitionState' is not defined)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Cellar/numpy/1.12.0/libexec/nose/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/local/Cellar/numpy/1.12.0/libexec/nose/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/local/Cellar/numpy/1.12.0/libexec/nose/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/Users/mattjohnson/RMGCODE/RMG-Py/rmgpy/cantherm/test_data/methoxy.py", line 13, in <module>
    transitionState(
NameError: name 'transitionState' is not defined

File: examples/cantherm/species/Benzyl/benzyl.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
```
Comment (mjohnson541) [inline examples/cantherm/species/Benzyl/benzyl.py]: This should be fixed

File: rmgpy/cantherm/statmech.py [hunk 3]
```diff
@@ -393,13 +393,6 @@ def load(self):
                        
             logging.debug('    Determining frequencies from reduced force constant matrix...')
             frequencies = numpy.array(projectRotors(conformer, F, rotors, linear, TS))
-            
-            # The frequencies have changed after projection, hence we need to recompute the ZPE
-            # We might need to multiply the scaling factor to the frequencies 
-            ZPE = self.getZPEfromfrequencies(frequencies)
```
Comment (mjohnson541) [inline rmgpy/cantherm/statmech.py]: This should be fixed now

**Comment by KEHANG** [comment]
Final comment: squash `752c629aba` into `cccc945`, and also squash `c471324` into `afecde8`

**Comment by mjohnson541** [comment]
Ok, not really sure how I ended up squashing the first one into the second to last one instead of the other way around as I intended, it didn't look like this in my local commit log, but I think this should be fine?

**Comment by KEHANG** [comment]
Last commit seems still containing the file we want to get rid of: `.DS_Store`.

**Comment by mjohnson541** [comment]
Ok, that should be fixed now.

**Pull request merged**